src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BytesKmerizer.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Iterator over successive Kmers from a byte[].
+ * Bytes are interpreted as ASCII characters.
+ * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
+ *
+ * Created by tsharpe on 12/18/15.
+ */
+public class BytesKmerizer implements Iterator<Kmer>
+{
+    private final byte[] seq;
+    private final int kSize;
+    private int idx;
+    private Kmer kmer;
+
+    public BytesKmerizer( final byte[] someSeq, final int someKSize )
+    {
+        seq = someSeq;
+        kSize = someKSize;
+        idx = 0;
+        kmer = new Kmer(kSize);
+        fill(0);
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+        return kmer != null;
+    }
+
+    @Override
+    public Kmer next()
+    {
+        if ( kmer == null ) throw new NoSuchElementException("Kmerization sequence exhausted.");
+        final Kmer result = kmer;
+        fill(kSize -1);
+        return result;
+    }
+
+    public static Kmer toKmer( final byte[] seq )
```

Comment:
Separation of concerns.  The Kmerization class has the knowledge of the character --> Base mapping.  To do what you suggest would cross-couple the two classes.  What happens when there's an alternate encoding of a byte[] that uses byte values 0, 1, 2, or 3 rather than 'A', 'C', 'G', or 'T'?  Now you're screwed.  Also, toKmer is unlikely to be used anywhere but in unit tests.  Maybe it should've been defined in the unit tests rather than here.